crates/relayer/src/chain/cosmos/query/tx.rs

```diff
@@ -183,6 +185,27 @@ pub async fn query_packets_from_block(
         }
     }
 
+    begin_block_events.append(
+        &mut block_results
+            .begin_block_events
+            .unwrap_or_default()
+            .into_iter()
+            .filter_map(|ev| filter_matching_event(ev, request, &request.sequences))
+            .map(|ev| IbcEventWithHeight::new(ev, height))
+            .collect(),
+    );
+
+    end_block_events.append(
+        &mut block_results
+            .end_block_events
+            .unwrap_or_default()
+            .into_iter()
+            .filter_map(|ev| filter_matching_event(ev, request, &request.sequences))
+            .map(|ev| IbcEventWithHeight::new(ev, height))
+            .collect(),
+    );
+    result.append(&mut begin_block_events);
+    result.append(&mut end_block_events);
```

Comment:
The order should be: begin_block_events, result, end_block_events.
Maybe also rename `result` to `tx_results`.

Reply:
8a59a565